### PR TITLE
misc: introduce PR labeler

### PR DESCRIPTION
**Pull request by lex111**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Let's try to automate a little bit the addition of labels to the opened PRs.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+version: 1
+labels:
+  - label: 'tag: new feature'
+    title: '^feat(\(.+\))?:.*'
+
+  - label: 'tag: bug fix'
+    title: '^fix(\(.+\))?:.*'
+
+  - label: 'tag: polish'
+    title: '^refactor(\(.+\))?:.*'
```
Comment (Josh-Cena) [inline .github/labeler.yml:10]: It's a bit complicated, but if you check the changelog, `feat` can be either "new feature" or "polish", and `fix` can be either "bug fix" or "polish" as well😅 `refactor`, on the other hand, can often be "maintenance" or "internal".

Our tags system is orthogonal to the semantic messages, and that's more or less intentional. I'm not against this change but just to illustrate the complication.

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    files:
+      - ^website/docs/.*
+      - ^website/versioned_docs/.*
+      - ^website/src/data/users.js
+
+  - label: 'tag: internal'
+    title: '^(chore|misc)(\(.+\))?:.*'
+  - label: 'tag: internal'
+    files:
+      - '^.github/*'
```
Comment (Josh-Cena) [inline .github/labeler.yml:24]: With the new "tag: maintenance" it's even more complicated. I think there's more human judgement here than any simple condition.

File: .github/labeler.yml (hunk 1 above)
Comment (lex111) [reply .github/labeler.yml:10]: Yes, but in most cases it should make work on manual labeling easier, at least a little.

File: .github/labeler.yml (hunk 2 above)
Comment (lex111) [reply .github/labeler.yml:24]: So, instead of `internal` we have to use `maintenance` now?

Comment (Josh-Cena) [reply .github/labeler.yml:24]: Let's keep it. "chore" and "misc" are more likely to be unlisted.

File: .github/labeler.yml (hunk 1 above)
Comment (Josh-Cena) [reply .github/labeler.yml:10]: That's true

**Comment by slorber** [comment]
I'd prefer to keep the tagging process manual for now.

When I see tags are missing, I'm more likely to assign a correct tag. 

If I see a PR is already tagged, I may not correct a wrong tag that was set automatically, and it is more likely to keep its wrong tag.

**Pull request closed without merging**